### ISSUE TITLE
False positive E1126: invalid-sequence-index from numpy.ufunc.reduce

### ISSUE THREAD

**Issue by d0sboots** [PyCQA/pylint#7702]
### Bug description

```py
#!/usr/bin/python
"""Demonstrate pylint issue"""

import numpy

print([0,1,2,3][numpy.bitwise_xor.reduce([8,9])])
```

The code runs fine, mypy reports no issues.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:6:6: E1126: Sequence index is not an int, slice, or instance with __index__ (invalid-sequence-index)

----------------------------------------------------------------------
Your code has been rated at -15.00/10 (previous run: -15.00/10, +0.00)
```

### Expected behavior

No lint issues.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.10.6 (main, Aug 10 2022, 11:19:32) [GCC 12.1.0]
```

### OS / Environment

Debian testing x86_64 GNU/Linux

### Additional dependencies

numpy=Debian version 1:1.21.5-1+b1 (1.21.5)
